lib/internal/bootstrap/pre_execution.js

```diff
@@ -164,6 +166,20 @@ function initializeReportSignalHandlers() {
   addSignalHandler();
 }
 
+function initializeHeapSnapshotSignalHandlers() {
+  const signal = getOptionValue('--heapsnapshot-signal');
+
+  if (!signal)
+    return;
+
+  require('internal/util').convertToValidSignal(signal);
```

Comment:
`signal = require('internal/util').convertToValidSignal(signal);`?

(And change it to `let` a few lines up.)

Reply:
Unfortunately, that converts the signal name to a number in addition to validating. That's what motivated https://github.com/nodejs/node/pull/27133#discussion_r273126030. The new validator will need to be used in at least one other place (`process.report` validation).

Reply:
Actually, I'm just going to add this new validator now to get it over with.